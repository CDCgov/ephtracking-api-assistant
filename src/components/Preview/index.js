import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import CodeBlock from '../CodeBlock';
import ApiJsonTree from '../../containers/ApiJsonTree';

/** Preview component that renders the API statement and the API JSON response  */
export class Preview extends Component {
  constructor(props) {
    super(props);
    /** track if API url was copied to clipboard */
    this.state = {
      copied: false
    };
  }

  componentWillReceiveProps(nextProps) {
    /** whenever the url changes, reset copied */
    if (this.props.url !== nextProps.url) {
      this.setState({
        copied: false
      });
    }
  }

  render() {
    const { url, validUrl } = this.props;
    const { copied } = this.state;
    return (
      <Fragment>
        <h5 className="title is-5">Output</h5>
        {/* Preview of API statement */}
        <div className="columns">
          <div className="column is-three-quarters">
            <CodeBlock>
              {/* only render as link if url is valid to avoid dead links */}
              {validUrl ? (
                <a target="_blank" rel="noopener noreferrer" href={url}>
                  {url}
                </a>
              ) : (
                <p>{url}</p>
              )}
            </CodeBlock>
          </div>
          <div className="column">
            <CopyToClipboard
              text={url}
              onCopy={() => this.setState({ copied: true })}
            >
              <button className="button" disabled={validUrl ? false : true}>
                {copied ? 'Copied' : 'Copy to Clipboard'}
              </button>
            </CopyToClipboard>
          </div>
        </div>
        {/* show JSON tree of API call if url is valid (complete) */}
        {validUrl && <ApiJsonTree url={url} />}
      </Fragment>
    );
  }
}

Preview.propTypes = {
  /** the API url to be used for the preview */
  url: PropTypes.string.isRequired,
  /** if valid, the component will call the API url for receiving data */
  validUrl: PropTypes.bool
};

Preview.defaultProps = {
  validUrl: true
};

export default Preview;
