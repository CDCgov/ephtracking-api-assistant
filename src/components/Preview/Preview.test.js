import React from 'react';
import Preview from './';
import { shallow } from 'enzyme';

describe('Preview Component', () => {
  const url = 'https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0';
  it('renders correctly', () => {
    const tree = shallow(<Preview url={url} isValid={false} />);
    expect(tree).toMatchSnapshot();
  });

  it('resets state.copied after receiving new url', () => {
    const newUrl =
      'https://ephtracking.cdc.gov/apigateway/api/v1/measures/242/0/0';
    const wrapper = shallow(<Preview url={url} />);
    expect(wrapper.state().copied).toBe(false);
    wrapper.setState({ copied: true });
    expect(wrapper.state().copied).toBe(true);
    wrapper.setProps({ url }); // same url, no change
    expect(wrapper.state().copied).toBe(true);
    wrapper.setProps({ url: newUrl }); // new url
    expect(wrapper.state().copied).toBe(false);
  });

  it('set state.copied to true after clicking on the copy-button', () => {
    const wrapper = shallow(<Preview url={url} />);
    expect(wrapper.state().copied).toBe(false);
    wrapper.instance().handleCopy();
    expect(wrapper.state().copied).toBe(true);
  });

  it('renders url as link only if valid is true', () => {
    const wrapper = shallow(<Preview url={url} validUrl={false} />);
    expect(wrapper.find({ href: url }).length).toBe(0);
    wrapper.setProps({ validUrl: true });
    expect(wrapper.find({ href: url }).length).toBe(1);
  });

  it('renders ApiJsonTree only if validUrl is set', () => {
    const wrapper = shallow(<Preview url={url} validUrl={false} />);
    expect(wrapper.find('ApiJsonTree').length).toBe(0);
    wrapper.setProps({ validUrl: true });
    expect(wrapper.find('ApiJsonTree').length).toBe(1);
  });
});
