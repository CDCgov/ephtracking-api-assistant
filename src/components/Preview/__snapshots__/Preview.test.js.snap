// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview
    isValid={false}
    url="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
    validUrl={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h5
          className="title is-5"
>
          Output
</h5>,
        <div
          className="columns"
>
          <div
                    className="column is-three-quarters"
          >
                    <CodeBlock>
                              <a
                                        href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                        rel="noopener noreferrer"
                                        target="_blank"
                              >
                                        https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                              </a>
                    </CodeBlock>
          </div>
          <div
                    className="column"
          >
                    <CopyToClipboard
                              onCopy={[Function]}
                              options={undefined}
                              text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                    >
                              <button
                                        className="button"
                                        disabled={false}
                              >
                                        Copy to Clipboard
                              </button>
                    </CopyToClipboard>
          </div>
</div>,
        <ApiJsonTree
          url="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Output",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Output",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="column is-three-quarters"
>
              <CodeBlock>
                            <a
                                          href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                          rel="noopener noreferrer"
                                          target="_blank"
                            >
                                          https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                            </a>
              </CodeBlock>
</div>,
            <div
              className="column"
>
              <CopyToClipboard
                            onCopy={[Function]}
                            options={undefined}
                            text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
              >
                            <button
                                          className="button"
                                          disabled={false}
                            >
                                          Copy to Clipboard
                            </button>
              </CopyToClipboard>
</div>,
          ],
          "className": "columns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CodeBlock>
                <a
                                href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                </a>
</CodeBlock>,
              "className": "column is-three-quarters",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <a
                  href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                  "href": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                "type": "a",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
>
                <button
                                className="button"
                                disabled={false}
                >
                                Copy to Clipboard
                </button>
</CopyToClipboard>,
              "className": "column",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="button"
                  disabled={false}
>
                  Copy to Clipboard
</button>,
                "onCopy": [Function],
                "options": undefined,
                "text": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Copy to Clipboard",
                  "className": "button",
                  "disabled": false,
                },
                "ref": null,
                "rendered": "Copy to Clipboard",
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h5
            className="title is-5"
>
            Output
</h5>,
          <div
            className="columns"
>
            <div
                        className="column is-three-quarters"
            >
                        <CodeBlock>
                                    <a
                                                href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                    >
                                                https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                                    </a>
                        </CodeBlock>
            </div>
            <div
                        className="column"
            >
                        <CopyToClipboard
                                    onCopy={[Function]}
                                    options={undefined}
                                    text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                        >
                                    <button
                                                className="button"
                                                disabled={false}
                                    >
                                                Copy to Clipboard
                                    </button>
                        </CopyToClipboard>
            </div>
</div>,
          <ApiJsonTree
            url="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Output",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Output",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="column is-three-quarters"
>
                <CodeBlock>
                                <a
                                                href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                >
                                                https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                                </a>
                </CodeBlock>
</div>,
              <div
                className="column"
>
                <CopyToClipboard
                                onCopy={[Function]}
                                options={undefined}
                                text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                >
                                <button
                                                className="button"
                                                disabled={false}
                                >
                                                Copy to Clipboard
                                </button>
                </CopyToClipboard>
</div>,
            ],
            "className": "columns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CodeBlock>
                  <a
                                    href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                    rel="noopener noreferrer"
                                    target="_blank"
                  >
                                    https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                  </a>
</CodeBlock>,
                "className": "column is-three-quarters",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <a
                    href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                    "href": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CopyToClipboard
                  onCopy={[Function]}
                  options={undefined}
                  text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
>
                  <button
                                    className="button"
                                    disabled={false}
                  >
                                    Copy to Clipboard
                  </button>
</CopyToClipboard>,
                "className": "column",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="button"
                    disabled={false}
>
                    Copy to Clipboard
</button>,
                  "onCopy": [Function],
                  "options": undefined,
                  "text": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Copy to Clipboard",
                    "className": "button",
                    "disabled": false,
                  },
                  "ref": null,
                  "rendered": "Copy to Clipboard",
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Preview Component rendes correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview
    isValid={false}
    url="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
    validUrl={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h5
          className="title is-5"
>
          Output
</h5>,
        <div
          className="columns"
>
          <div
                    className="column is-three-quarters"
          >
                    <CodeBlock>
                              <a
                                        href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                        rel="noopener noreferrer"
                                        target="_blank"
                              >
                                        https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                              </a>
                    </CodeBlock>
          </div>
          <div
                    className="column"
          >
                    <CopyToClipboard
                              onCopy={[Function]}
                              options={undefined}
                              text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                    >
                              <button
                                        className="button"
                                        disabled={false}
                              >
                                        Copy to Clipboard
                              </button>
                    </CopyToClipboard>
          </div>
</div>,
        <ApiJsonTree
          url="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Output",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Output",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="column is-three-quarters"
>
              <CodeBlock>
                            <a
                                          href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                          rel="noopener noreferrer"
                                          target="_blank"
                            >
                                          https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                            </a>
              </CodeBlock>
</div>,
            <div
              className="column"
>
              <CopyToClipboard
                            onCopy={[Function]}
                            options={undefined}
                            text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
              >
                            <button
                                          className="button"
                                          disabled={false}
                            >
                                          Copy to Clipboard
                            </button>
              </CopyToClipboard>
</div>,
          ],
          "className": "columns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CodeBlock>
                <a
                                href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                </a>
</CodeBlock>,
              "className": "column is-three-quarters",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <a
                  href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                  "href": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                "type": "a",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
>
                <button
                                className="button"
                                disabled={false}
                >
                                Copy to Clipboard
                </button>
</CopyToClipboard>,
              "className": "column",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="button"
                  disabled={false}
>
                  Copy to Clipboard
</button>,
                "onCopy": [Function],
                "options": undefined,
                "text": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Copy to Clipboard",
                  "className": "button",
                  "disabled": false,
                },
                "ref": null,
                "rendered": "Copy to Clipboard",
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h5
            className="title is-5"
>
            Output
</h5>,
          <div
            className="columns"
>
            <div
                        className="column is-three-quarters"
            >
                        <CodeBlock>
                                    <a
                                                href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                    >
                                                https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                                    </a>
                        </CodeBlock>
            </div>
            <div
                        className="column"
            >
                        <CopyToClipboard
                                    onCopy={[Function]}
                                    options={undefined}
                                    text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                        >
                                    <button
                                                className="button"
                                                disabled={false}
                                    >
                                                Copy to Clipboard
                                    </button>
                        </CopyToClipboard>
            </div>
</div>,
          <ApiJsonTree
            url="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Output",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Output",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="column is-three-quarters"
>
                <CodeBlock>
                                <a
                                                href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                >
                                                https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                                </a>
                </CodeBlock>
</div>,
              <div
                className="column"
>
                <CopyToClipboard
                                onCopy={[Function]}
                                options={undefined}
                                text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                >
                                <button
                                                className="button"
                                                disabled={false}
                                >
                                                Copy to Clipboard
                                </button>
                </CopyToClipboard>
</div>,
            ],
            "className": "columns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CodeBlock>
                  <a
                                    href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                                    rel="noopener noreferrer"
                                    target="_blank"
                  >
                                    https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
                  </a>
</CodeBlock>,
                "className": "column is-three-quarters",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <a
                    href="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0
</a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                    "href": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CopyToClipboard
                  onCopy={[Function]}
                  options={undefined}
                  text="https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0"
>
                  <button
                                    className="button"
                                    disabled={false}
                  >
                                    Copy to Clipboard
                  </button>
</CopyToClipboard>,
                "className": "column",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="button"
                    disabled={false}
>
                    Copy to Clipboard
</button>,
                  "onCopy": [Function],
                  "options": undefined,
                  "text": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Copy to Clipboard",
                    "className": "button",
                    "disabled": false,
                  },
                  "ref": null,
                  "rendered": "Copy to Clipboard",
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/measures/null/0/0",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
